libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts

```diff
@@ -3,17 +3,96 @@
 import {
     AnalyticalWidgetType,
     IInsight,
+    IInsightDefinition,
+    IKpi,
     isDashboardWidget,
     ISettings,
+    isInsight,
+    isKpi,
     isKpiWidget,
+    isKpiWithoutComparison,
     IWidget,
     widgetType,
 } from "@gooddata/sdk-model";
-import { fluidLayoutDescriptor } from "@gooddata/sdk-ui-ext";
+import {
+    fluidLayoutDescriptor,
+    getInsightSizeInfo,
+    INSIGHT_WIDGET_SIZE_INFO_DEFAULT,
+    INSIGHT_WIDGET_SIZE_INFO_DEFAULT_LEGACY,
+    IVisualizationSizeInfo,
+    KPI_WIDGET_SIZE_INFO_DEFAULT,
+    KPI_WIDGET_SIZE_INFO_DEFAULT_LEGACY,
+} from "@gooddata/sdk-ui-ext";
+
+import { ObjRefMap } from "../metadata/objRefMap";
+
+/**
+ * @internal
+ */
+export type MeasurableWidgetContent = IInsightDefinition | IKpi;
+
+/**
+ * @internal
+ */
+export function getSizeInfo(
+    settings: ISettings,
+    widgetType: AnalyticalWidgetType,
+    widgetContent?: MeasurableWidgetContent,
+): IVisualizationSizeInfo {
+    if (widgetType === "kpi") {
+        return getKpiSizeInfo(settings, widgetContent);
+    }
 
-import { getSizeInfo, MeasurableWidgetContent } from "../../_staging/layout/getSizeInfo";
-import { ObjRefMap } from "../../_staging/metadata/objRefMap";
+    return getVisualizationSizeInfo(settings, widgetContent);
+}
+
+function getVisualizationSizeInfo(
+    settings: ISettings,
+    insight?: MeasurableWidgetContent,
+): IVisualizationSizeInfo {
+    let sizeInfo;
+    if (isInsight(insight)) {
+        sizeInfo = getInsightSizeInfo(insight, settings);
+    }
+
+    if (!sizeInfo) {
+        if (!settings.enableKDWidgetCustomHeight) {
+            return INSIGHT_WIDGET_SIZE_INFO_DEFAULT_LEGACY;
+        }
+        return INSIGHT_WIDGET_SIZE_INFO_DEFAULT;
+    }
+    return sizeInfo;
+}
+
+function getKpiSizeInfo(settings: ISettings, kpi?: MeasurableWidgetContent): IVisualizationSizeInfo {
+    if (!settings.enableKDWidgetCustomHeight) {
+        return KPI_WIDGET_SIZE_INFO_DEFAULT_LEGACY;
+    }
+    if (!isKpi(kpi)) {
+        return KPI_WIDGET_SIZE_INFO_DEFAULT;
+    }
+    return {
+        width: {
```

Comment:
I saw these in the previous PR, can we extract the values to some `constats.ts` file to make it more maintainable?